gatsby-node.js

```diff
@@ -401,21 +411,64 @@ const getLearnPageFilters = async () => {
     return filters;
 };
 
-exports.onCreatePage = async ({ page, actions }) => {
-    if (page.path === '/learn/') {
-        const { createPage, deletePage } = actions;
-        const allArticles = await getLearnPageArticles();
-        const featuredArticles = getFeaturedLearnArticles(allArticles);
-        const filters = await getLearnPageFilters(allArticles);
-        deletePage(page);
-        createPage({
-            ...page,
-            context: {
-                ...page.context,
-                allArticles,
-                featuredArticles,
-                filters,
-            },
-        });
+const getFeaturedArticlesForPage = async (
+    page,
+    defaultFeaturedArticles,
+    allArticles = null
+) => {
+    if (!allArticles) {
+        allArticles = await getLearnPageArticles();
+    }
+    // TODO: Verify field names with docsp
+    const featuredArticleSlugs =
+        getNestedValue(['featuredArticles', `${page}Page`], metadata) ||
```

Comment:
Should we move to `dlv` more consistently?

Reply:
This might be too much of a nit, but `${page}Page` will shoe-horn us into that format, a map might be better here.

Reply:
> Should we move to `dlv` more consistently?

Yeah I was curious about this, since it is almost equivalent to getNestedValue. I wouldn't mind using `dlv` since we can specify a default without having to `|| <defaultValue>` which feels a bit. more hacky.

What do you think?

Reply:
Yep, I think that makes sense. Technically we could use optional changing which is built-in, but... then we'd have 3 ways to do this.

One thing to watch for when making this change is if we're getting a falsy value right now that then uses `||` to fallback. `dlv` I believe only fallsback when a value or key is undefined (like default params)